### PR TITLE
GUI

### PR DESCRIPTION
**Pull request by Munawwar**
Use pywebview, which uses the OS'es installed browser

I wanted to also package it to independent desktop app,.. but I could not get pyinstaller to work with torch though

```
pip3 install pyinstaller
pyinstaller -y --app-data index.html:. --app-data models:/models:/ app.py
```

### CODE REVIEW

**Comment by Munawwar** [comment]
An unrelated thing to mention.. maybe the background cutout at the edges could be improved by using the neural net's output as a mask/matt to grabcut algorithm?

**Comment by Munawwar** [comment]
Here is my approach so far to improve the edges:

```python
import cv2
...
...

model = model_detect("u2net")
image = model.process_image(input_path)
mask = np.array(image.getchannel('A'))

# create a better mask
mask[mask < 128] = 0
mask[mask >= 128] = 255
# duplicate it to 3 color channels
mask = cv2.cvtColor(mask, cv2.COLOR_GRAY2RGB)

src = cv2.imread(input_path, 1)

# ------- do an alpha blended cutout ---------
# first by blurring the borders
# TODO: make erode/blur scale with image scale?
mask2 = cv2.erode(mask, np.ones((15,15),np.uint8) , iterations = 1)
mask3 = cv2.GaussianBlur(mask2, (15,15), 0)
alpha = mask3.astype(float) * 1.2  # making blend stronger
alpha[alpha > 255] = 255.0
# Normalize the alpha mask to keep intensity between 0 and 1
alpha = alpha / 255.0

# now.. cutting time!
foreground = np.copy(src).astype(float)
# Multiply the foreground with the alpha matte
foreground = cv2.multiply(alpha, foreground)

background = np.ones_like(foreground, dtype=float) * 255.0
# Multiply the background with ( 1 - alpha )
background = cv2.multiply(1.0 - alpha, background)

# Add the masked foreground and background.
finalImage = cv2.add(foreground, background)
finalImage[finalImage > 255] = 255
finalImage = finalImage.astype(np.uint8)
```

Shall I add this as a CLI option?

**Comment by OPHoperHPO** [comment]
Thank you for your contribution to the development of the project! I slightly improved the look of your GUI by adding animations and Material design there.

**Comment by OPHoperHPO** [comment]
I tried the procedure of packing the program into a file using pyinstaller and the command you provided.
Unfortunately, there is something wrong with the arguments for pyinstaller.
I tried to change them a little bit, but still it gives me an error.
Pyinstaller crashes when trying to pack models but not pytorch.
I used this command
`pyinstaller -y --add-data ./gui/index.html:gui/index.html --add-data ./models/:models/ gui.py`

**Comment by OPHoperHPO** [comment]
> Here is my approach so far to improve the edges:
> ...
> Shall I add this as a CLI option?

Your method of improving borders is good, but sometimes it cuts borders too much. As an example, I cited the work of your and my very slow algorithm (u2net + fastrcnn) at the alpha stage, based on detecting the boundaries of objects in the photo and separately removing the background from each object in the photo. [Here's the link](https://colab.research.google.com/gist/OPHoperHPO/6f91f2a2d5f08772f1cea9db77138046/rm_bg_fastrcnn_alpha_cv2_test.ipynb). Yes, you can add an argument(`-pp <name_of_your_method>`), but do not embed the processing algorithm directly in main.py. Put the processing algorithm as a child of the `Postprocessing` class in `libs/postprocessing.py`.

**Comment by Munawwar** [comment]
@OPHoperHPO Can you replace
```
# TODO: make erode/blur scale with image scale?
mask2 = cv2.erode(mask, np.ones((15,15),np.uint8) , iterations = 1)
mask3 = cv2.GaussianBlur(mask2, (15,15), 0)
```

```
from math import floor
...

# find a blur/erode kernel size that scales with the image
# take it as 1% of the width/height (it has to be an odd number >= 3)
kernelSize = floor(min(mask.shape[0], mask.shape[1]) * 0.01)
kernelSize = max((kernelSize - (kernelSize % 2) + 1), 3)

mask2 = cv2.erode(mask, np.ones((kernelSize,kernelSize),np.uint8) , iterations = 1)
mask3 = cv2.GaussianBlur(mask2, (kernelSize,kernelSize), 0)
```
